### ISSUE TITLE
The port to 1.9!

### ISSUE THREAD

**Issue by skyem123** [Victorious3/Integrated-Circuits#183]
So... we might as well try to do it, because BTM will happen soon!

@Inari-Whitebear is trying to do this, and I will try to help them.

This issue is for us to keep track of stuff, etc.
Sadly, I cannot assign Inari to this issue, so I'm not assigning anyone to this issue.

Things to do:
- [ ] Get it to compile
- [ ] Get it to not crash while starting
- [ ] Get it to somewhat work in world
- [ ] Get it to work
- [ ] Re-add removed stuff

**Comment by skyem123** [comment]
This is a list of stuff that I have removed, that should be rewritten for some reason:
- [ ] dae272792b07061343e7dc47f3b768d47cd58249 - MultiPart stuff. Should be rewritten to fit the new system. Removed because it's not a core feature.
- [ ] 466c961ca306e4495e4bb7f8099352d089fccd9b - Computer Peripheral stuff. should be rewritten, as the old code was very buggy and broken. ~~Also, ComputerCraft doesn't exist for Minecraft 1.9~~
- [ ] 54bfc9b33c70002d570cf52018d38c1ea5bede20 - NEI Plugin. Should be replaced with a JEI Plugin.
- [ ] 522ba128b4175f1f9186b55be6db39ec88bfdf6a - ProjectRed stuff. Should be replaced with stuff for Charset.

**Comment by Vexatos** [comment]
http://www.computercraft.info/forums2/index.php?/topic/27036-computercraft-180-beta-information-minecraft-194/

**Comment by skyem123** [comment]
Oh...  oops.

**Comment by skyem123** [comment]
```
<Skye> Vu`iku, I'm helping by removing large parts of the code
<Vu`iku> Skye, well remove the entire API garbage and compatibility junk
<Vu`iku> And don't let them solder anymore, just place directly
<Skye> so everything in the API package?
<Vu`iku> Not everything
<Vu`iku> But it has to be rewritten, and the asm stuff needs to die
<Vu`iku> Oh and no text rendering on items
<Skye> the main issue is that codechickenlib isn't for 1.9
<Vu`iku> Try contacting that person who did the video on ICs
<Vu`iku> He wrote an addon for it so he'd probably make a good maintainer 
<Skye> so I should remove all the code that talks to other mods?
<Vu`iku> That too
<Vu`iku> But really, finding that person and making him join irc is probably easier
```

**Comment by Inari-Whitebear** [comment]
- [ ] Inari-Whitebear/Integrated-Circuits@fa25ac5 deleted:    src/main/java/moe/nightfall/vic/integratedcircuits/api/gate/IGatePeripheralProvider.java
  maybe reimplement sometime

**Comment by Inari-Whitebear** [comment]
So it compiles and starts. You can place some blocks, but nothing really works yet aside opening the Assembler GUI

**Comment by Inari-Whitebear** [comment]
Next things to fix: Mapping on CAD Assembler is wrong (North IO Pin doesn't activate North IO Pin etc), Cables can't seem to communicate with IO Pins regardless.

**Comment by colstrom** [comment]
Just wanted to say Thanks for the effort on this so far. I know it's still work in progress, but I'm looking forward to being able to play with this.

**Comment by Xcallono** [comment]
Yeah, I've got a problem


You've stopped porting :(

**Comment by skyem123** [comment]
Yeah... It's just not easy to port such a mod and neither me nor Vic are interested in Minecraft anymore.
Also Vic's code uses some complex dark java art that I don't understand. 😛
I'm busy with life and school and so is Vic. It's open source so if you can get someone else to do it, I'll happily review any pull request.

Sorry, I know it sucks, but it's actually a project too big for me right now (I've got something I need to do for school, for example).

**Comment by Inari-Whitebear** [comment]
Too much work, especially it's not only understanding MC modding, but also understanding the original Coder's thougths xD Not much interested in MC or MC modding anymore, and theres SCM now, which I consider superior anyway.

**Comment by Inari-Whitebear** [comment]
Oh and sometimes the original coder is oddly specific about their likes and dislikes ;)

**Comment by Xcallono** [comment]
Super Circuit Maker is odd and doesn't let you save PCB designs, something I feel is wrong. Furthermore, the creator hasn't shown support for a while to the mod.

**Comment by Inari-Whitebear** [comment]
He's updating it to a newer MC version, and it'll be part of a bigger mod later on.

And it does let you save your circuit designs

**Comment by Victorious3** [comment]
You can fork and do whatever you want with the source. The complicated part that nobody wants to work on is rendering, as all dependencies are gone and what they got now is an overcomplicated clusterfuck which is no fun to work with at all. If you want to see this ported you have to do it yourself, I'm afraid. I'll happily give answer about my original intent tho ^^
